### PR TITLE
Fix for instantiation of pulled dataset

### PR DESCRIPTION
**Pull request by ilongin**
Adding a fix for instantiated datasets pulled from Studio.
In addition, it adds small improvement for avoiding going to Studio if dataset with specified version is already present locally

### CODE REVIEW

**Review by shcheklein — COMMENTED** [review]
Thanks @ilongin ! Can we add a test for this?

**Review by shcheklein — COMMENTED** [review]
Btw, do we have any docs? what is the command that was broken?

File: src/datachain/catalog/catalog.py [hunk 1]
```diff
@@ -1350,6 +1355,13 @@ def _instantiate_dataset():
             # we will create new one if it doesn't exist
             pass
 
+        if dataset and version and dataset.has_version(version):
+            """No need to communicate with Studio at all"""
```
[the remark this answers is not in the document]
Comment (mattseddon) [reply src/datachain/catalog/catalog.py:1359]: E.g. `my_dataset@v1` on remote is full of PDFs and `my_dataset@v1` locally is images.

[the remark this answers is not in the document]
Comment (ilongin) [reply src/datachain/catalog/catalog.py:1359]: Good question. No, there is no such thing ATM. Currently it's up to user to make sure he doesn't "shadow" remote dataset. In order to make this kind of validation we need to have `equal` function between datasets (I will create an issue for this) and my first feeling is that it's not that trivial to implement it (maybe I'm wrong, idk)

**Comment by ilongin** [comment]
> Thanks @ilongin ! Can we add a test for this?

Added a test

**Comment by ilongin** [comment]
> Btw, do we have any docs? what is the command that was broken?

Command that was broken was `datachain pull ds://some-dataset-from-studio`  . By default this command pulls dataset to local DB but it also starts instantiation of it and that instantiation was broken. Instantiation can be avoided by adding `--no-cp` flag.
We can maybe think of removing instantiation as default option and add it as optional (to have `--cp` flat instead of `--no-cp`).

So currently if user wants to pull dataset and instantiate it locally, it can be done in two ways:

1. 
```
datachain pull ds://my-bucket -o output_dir  # does pulling and instantiating in one command
```

2.
```
datachain pull ds://my-bucket --no-cp  # does only pulling
datachain cp ds://my-bucket output_dir  # instantiating local dataset
```

**Comment by shcheklein** [comment]
> We can maybe think of removing instantiation as default option and add it as optional (to have --cp flat instead of --no-cp).

sounds reasonable to me. Or do a separate command to instantiate it?

File: src/datachain/catalog/catalog.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ilongin) [reply src/datachain/catalog/catalog.py:1359]: Actually, it seems like a simple uuid should be enough for this, which is trivial

**Review by shcheklein — APPROVED** [review]

**Pull request merged**